src/components/quiz/QuizQuestions.tsx

```diff
@@ -1,18 +1,91 @@
-import React from "react";
+import React, { useState } from "react";
 import { Question } from "./questions";
-import { Reward } from "./Reward";
+import { UserDetailsForm } from "./UserDetailsForm";
+import './QuizQuestions.scss';
 
 interface QuizQuestionsProps {
-  questions: Question[];
+   questions: Question[];
+   score: number;
+   setScore: (score: number) => void;
+   isEndOfQuestion: boolean;
+   setIsEndOfQuestion: (isEndOfQuestion: boolean) => void;
 }
 
+
 export const QuizQuestions: React.FunctionComponent<QuizQuestionsProps> = ({
-  questions,
+  questions,score,setScore,isEndOfQuestion,setIsEndOfQuestion
 }) => {
+
+  const [activeQuestionIndex, setActiveQuestionIndex] = useState(0);
+  const activeQuestion = questions[activeQuestionIndex];
+  const [activeQuestionClicked, setActiveQuestionClicked] = useState(false);
+  const [message, setMessage] = useState("");
+  
+  const selectAnswer = (answerIndex: number) => {
+    setActiveQuestionClicked(true);
+    if (answerIndex === activeQuestion.correctAnswerIndex) {
+      setScore(score + 1);
+      setMessage('Correct!!!')
+    } else{
+      setMessage('Incorrect!!!')
+    } 
+
+    isQuizFinished();
+  }
+
+  const isQuizFinished = ()=>{
+    if(activeQuestionIndex === 10)
```

Comment:
Space:
```suggestion
    if (activeQuestionIndex === 10)
```
and don't hard-code "10"! Get the length from the list:
```suggestion
    if(activeQuestionIndex === questions.length - 1)
```
(we subtract 1 because the indexing is from 0)